src/tests/ftest/launch.py

```diff
@@ -1733,6 +1748,115 @@ def setup_test_files(self, args, yaml_dir):
             # Collect the host information from the updated test yaml
             test.set_yaml_info(args.include_localhost)
 
+    @staticmethod
+    def _query_create_group(hosts, group, create=False):
+        """Query and optionally create a group on remote hosts.
+
+        Args:
+            hosts (NodeSet): hosts on which to query and create the group
+            group (str): group to query and create
+            create (bool, optional): whether to create the group if non-existent
+
+        Raises:
+            LaunchException: if there is an error querying or creating the group
+
+        Returns:
+            str: the group's gid
+
+        """
+        # Query the group and get the gid
+        logger.info('Querying group %s', group)
+        result = getent(logger, hosts, 'group', group)
+        if result.passed:
+            return result.output[0].stdout[0].split(':')[2]
```

Comment:
I'd be careful with assuming the expected output is always on the first line, e.g., `stdout[0]`.  A regex here may be safer.  Also note that `result.passed` does not mean that all the hosts returned the same output, that's what `result.homogeneous` indicates.

Reply:
Fixed